src/request.test.ts

```diff
@@ -0,0 +1,53 @@
+import axios from "axios";
+import MockAdapter from "axios-mock-adapter";
+
+import { SkynetClient } from ".";
+import { ExecuteRequestError } from "./request";
+import { defaultSkynetPortalUrl } from "./utils/url";
+
+const portalUrl = defaultSkynetPortalUrl;
+const client = new SkynetClient(portalUrl);
+const skylink = "XABvi7JtJbQSMAcDwnUnmp2FKDPjg8_tTTFP4BwMSxVdEg";
+
+describe("ExecuteRequestError", () => {
+  let mock: MockAdapter;
+
+  beforeEach(() => {
+    mock = new MockAdapter(axios);
+    mock.resetHistory();
+  });
+
+  it("should have a working instanceof and also be an axios error", async () => {
+    try {
+      // Should result in a 404 as we haven't set up any mock handlers.
```

Comment:
nit: Wouldn't it be clearer if we specifically mocked it to return a 404?

Reply:
I can do that for clarity. 👍